### PR TITLE
feat: validate input media type for `oras.PackManifest `

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Resolves: #571

### CODE REVIEW

File: pack.go [hunk 1]
```diff
@@ -369,6 +384,22 @@ func pushManifest(ctx context.Context, pusher content.Pusher, manifest any, medi
 	return manifestDesc, nil
 }
 
+// pushCustomConfig generates and pushes a custom config blob.
+func pushCustomConfig(ctx context.Context, pusher content.Pusher, mediaType string, annotations map[string]string) (ocispec.Descriptor, error) {
```
Comment (shizhMSFT) [inline pack.go:388]: It is not custom config but legacy empty config.

File: pack.go [hunk 2]
```diff
@@ -190,28 +199,30 @@ func packArtifact(ctx context.Context, pusher content.Pusher, artifactType strin
...
-		// push config
-		if err := pushIfNotExist(ctx, pusher, configDesc, configBytes); err != nil {
-			return ocispec.Descriptor{}, fmt.Errorf("failed to push config: %w", err)
+		if artifactType != "" {
+			if err := validateMediaType(artifactType); err != nil {
+				return ocispec.Descriptor{}, fmt.Errorf("invalid artifactType format: %s: %w", artifactType, err)
+			}
+		} else {
+			artifactType = MediaTypeUnknownConfig
+		}
```
Comment (shizhMSFT) [inline pack.go:221]:
```suggestion
		if artifactType == "" {
			artifactType = MediaTypeUnknownConfig
		} else if err := validateMediaType(artifactType); err != nil {
			return ocispec.Descriptor{}, fmt.Errorf("invalid artifactType format: %s: %w", artifactType, err)
		}
```

File: pack.go [hunk 3]
```diff
@@ -392,3 +423,11 @@ func ensureAnnotationCreated(annotations map[string]string, annotationCreatedKey
 	copied[annotationCreatedKey] = now.Format(time.RFC3339)
 	return copied, nil
 }
+
+// validateMediaType validates the format of mediaType.
+func validateMediaType(mediaType string) error {
+	if !mediaTypeRegexp.MatchString(mediaType) {
+		return errdef.ErrInvalidMediaType
```
Comment (shizhMSFT) [inline pack.go:430]: It seems can be refactored as 
```suggestion
		return fmt.Errorf("%s: %w", mediaType, errdef.ErrInvalidMediaType)
```

File: pack.go (hunk 1 above)
Comment (Wwwsylvia) [reply pack.go:388]: maybe `customEmptyConfig` 😀

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**